File: src/pub_and_sub/scripts/sub.py
import rospy
from std_msgs.msg import String
from pub_and_sub.msg import Person

def doMsg(p):
    rospy.loginfo(f"订阅的数据是: %s %d %f", p.name, p.age, p.height)


if __name__ == '__main__':
    rospy.init_node("sub")
    sub = rospy.Subscriber("pub_theme", Person, doMsg, queue_size=10)

    rospy.spin()

File: src/pub_and_sub/src/sub.cpp
#include"ros/ros.h"
#include"std_msgs/String.h"
#include<sstream>
#include"pub_and_sub/Person.h"
void revMsg(const pub_and_sub::Person::ConstPtr &msg_ptr)
{
    ROS_INFO("接收：%s %d %.2f", msg_ptr->name.c_str(), msg_ptr->age, msg_ptr->height);
}

int main(int argc, char *argv[])
{
    setlocale(LC_ALL,"");
    ros::init(argc, argv, "subscribe");
    ros::NodeHandle handle;
    ros::Subscriber sub = handle.subscribe("pub_theme", 10, revMsg);

    ros::spin();
    return 0; 
}